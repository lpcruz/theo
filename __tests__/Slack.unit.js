'use strict';
const Slack = require('../src/API/Slack');

const fakeChannel = {
  NAME: '#dev',
  CHANNEL_ID: 'C0145H2QG74',
  SLACK_HOOK: 'foobarbazhook1234'
}

describe('Slack', () => {
  let request;
  let slack;
  
  beforeEach(() => {
    request = jest.fn();
    slack = new Slack(request);
  });
  
  describe('greet', () => {
    it('should wave back', () => {
      const message = {
        message: {
          user: 'dev',
          channel: fakeChannel.CHANNEL_ID
        }
      };
      expect(slack.greet(message)).toMatchSnapshot();
    });
  });
  
  describe('help', () => {
    it('should provide instructions', () => {
      const message = {
        message: {
          user: 'dev',
          channel: fakeChannel.CHANNEL_ID
        }
      };
      expect(slack.greet(message)).toMatchSnapshot();
    });
  });
});