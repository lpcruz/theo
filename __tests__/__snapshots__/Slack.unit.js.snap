// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slack greet should wave back 1`] = `
Array [
  Object {
    "text": Object {
      "text": ":wave: <@dev>!",
      "type": "mrkdwn",
    },
    "type": "section",
  },
]
`;

exports[`Slack help should provide instructions 1`] = `
Array [
  Object {
    "text": Object {
      "text": ":wave: <@dev>!",
      "type": "mrkdwn",
    },
    "type": "section",
  },
]
`;
